### ISSUE TITLE
Files are re-uploaded after reboot

### ISSUE THREAD

**Issue by rtfmjoey** [nextcloud/android#756]
### Actual behaviour
I uploaded all my camera pictures from my phone, but after a reboot of the phone the upload starts all over again. Files get the suffix (2) and (3) after upload as they already exist.

It doesn't matter if the file is kept in the original folder moved to the app folder. 

Files from the DCIM folder seem to be copied to the app folder and are not deleted from the original folder.

Perhaps NC doesn't have enough permissions or something like that?

### Expected behaviour
Files should be moved, duplicates should be detected.
 
### Steps to reproduce
1. Select camera folder for upload
2. Wait till upload completes
3. Reboot phone
4. Profit!

### Environment data
Android version:
6.0.1

Device model: 
Huawei GX8

Stock or customized system:
Stock B421 (Latest)

Nextcloud app version:
1.4.2 from Play store

Nextcloud server version:
11.0.2

#### Web server error log
N/A
Insert your webserver log here
N/A

#### Nextcloud log (data/nextcloud.log)
App has been removed from phone, will retry and post the logs.

**Comment by MorrisJobke** [comment]
cc @mario @AndyScherzinger @tobiasKaminsky

**Comment by mario** [comment]
@MorrisJobke aware of this. Dunno even in theory how this can happen and also very hard to replicate without a phone...

**Comment by asychev** [comment]
Same for me, Huawei P9 Lite, android 6.0.1. Messed my photo gallery...

**Comment by mario** [comment]
@asychev still investigating :-/ Hard without a phone that has that behaviour....

**Comment by asychev** [comment]
Can I do something to help? Huawei P9, android 7 - same bug...

**Comment by Wikinaut** [comment]
Ugly bug, exactly the same problem here. Please fix with high priority.

Huawei P8 Lite, Android 6.0

**Comment by mario** [comment]
We can only fix if someone is willing to install Android studio with all SDKs + master tarball, Vysor, connect phone to the computer, and let me access via TeamViewer since we don't have the phone...

And nobody seems to have the time to do it..

**Comment by Wikinaut** [comment]
Added: the device has (inter alia) two folders:

* /storage/1234-1234/DCIM/Camera and
* /storage/emulated/0/DCIM/Camera

for some reasons, I do not understand, the content is different, both have photos from the last weeks.

**Comment by mario** [comment]
One is internal, one external (SD card) storage afaik.

**Comment by Wikinaut** [comment]
@asychev @rtfmjoey to clean up the mess (duplicate and n-licate pictures in your cloud upload area), I recommend FSLint.

**Comment by Wikinaut** [comment]
@asychev @rtfmjoey please can you report here whether a de- and re-installation of the Nextcloud app could help to fix the problem.

**Comment by mario** [comment]
@Wikinaut do you have the time to do the dance I mentioned above?

"We can only fix if someone is willing to install Android studio with all SDKs + master tarball, Vysor, connect phone to the computer, and let me access via TeamViewer since we don't have the phone..."

**Comment by Wikinaut** [comment]
I do not allow TeamViewer.

**Comment by asychev** [comment]
I upgraded my Huawei P9 lite to android 7.0 and need to go to another issue, where autoupload not working ¯ \ _ (ツ) _ / ¯

**Comment by mario** [comment]
@asychev can you do what I asked above?

**Comment by asychev** [comment]
@mario I think yes, but for now reupload issue gone for me, as I can see. Can we contact through telegram and discuss?

**Comment by mario** [comment]
@asychev you have other issues now though :P Sure, poke me at mario@nextcloud.com via mail

**Comment by asychev** [comment]
@mario check you mailbox

**Comment by mario** [comment]
This issue is now considered fixed and will be in 1.5. Thank you @asychev for your help.

**Comment by MorrisJobke** [comment]
> This issue is now considered fixed and will be in 1.5. Thank you @asychev for your help.

That means we can close this one here? :)

**Comment by mario** [comment]
@MorrisJobke not until I can get this merged where it needs to be :P

**Comment by asychev** [comment]
Happy to help, @mario. Please, release it asap (at least as RC/beta), app just not unusable for now.

**Comment by mario** [comment]
Dear users,

thank you for your patience. After lots of work, we’re super excited to be able to give you a test version with new auto upload implementation. Keep in mind we will only accept bugs related to Auto Upload in this APK, as other things might be broken.

https://cloud.nextcloud.com/s/ftZraTruRQjTe0R

This is a dev APK so it will not overwrite your existing play store/f-droid APK unless you’re already running dev version. Also, you might need to remove your existing dev version before installing this one.

Explanation of the current functionality:
- added support for custom folders
- separated image and video folders
- checks for new files happens every 15-20 minutes
- automatic retry every 15-20 minutes or when you force close/open the app
- automatic pause once network is changed (untested!)
- significantly less battery usage

All and any testing is more than appreciated, especially with large and ever-going files like Videos. (like, try to record a 30 min video).

Let me know if you’ve got any questions.

Cheers,
Mario

**Comment by nille02** [comment]
Nice that its fixed, i had this issue some time now and looked for a solution. 

I see that there are 2.0 Release Candidates available, but why not upload them to the play store as a beta release?

**Comment by mario** [comment]
They are a beta releAse there?:)

**Comment by nille02** [comment]
Your right, at the bottom was the Beta Button. My bad, and thank for this nice app.

**Comment by Wikinaut** [comment]
ty; yes, seems to work now.
